src/frontends/tensorflow/src/op/queue_dequeue.cpp

```diff
@@ -16,50 +16,41 @@ namespace ov {
-    // QueueDequeue operation generates multiple outputs
-    // which we prune and create Parameter node for
-    vector<string> supported_ops = {"QueueDequeue",
-                                    "QueueDequeueV2",
-                                    "QueueDequeueUpTo",
-                                    "QueueDequeueUpToV2",
-                                    "QueueDequeueMany"};
-    default_op_checks(node, 2, supported_ops);
+OutputVector translate_queue_dequeue_base(const ov::frontend::tensorflow::NodeContext& node,
+                                          const Output<Node>& handle,
+                                          const Dimension& batch_dim) {
     auto node_name = node.get_name();
-    auto handle = node.get_input(0);
-    auto n = node.get_input(1);
-
-    // compute batch dimension for outputs
-    // this is a number of batch objects emitted from QueueDequeue
-    Dimension batch_dim = Dimension::dynamic();
-    if (auto n_const = get_constant_from_source(n)) {
-        auto n_value = n_const->cast_vector<int64_t>();
-        if (n_value.size() > 0 && n_value[0] > 0) {
-            batch_dim = n_value[0];
-        }
-    }
-
     // all data about output shapes and types are saved in FIFOQueue operation
     auto fifo_queue = as_type_ptr<FIFOQueue>(handle.get_node_shared_ptr());
     TENSORFLOW_OP_VALIDATION(
         node,
         fifo_queue,
         "[TensorFlow Frontend] Internal error: only FIFOQueue is supported as a producer for QueueDequeue operation.");
-    auto output_shapes = fifo_queue->get_component_shapes();
     auto output_types = fifo_queue->get_component_types();
-    size_t output_num = output_shapes.size();
-    TENSORFLOW_OP_VALIDATION(
-        node,
-        output_num == output_types.size(),
-        "[TensorFlow Frontend] Incorrect input model: lenghts of output_types and output_shapes do not match.");
+    size_t output_num = output_types.size();
+
+    // output shapes is optional for FIFOQueue
+    // by default this is empty list so we should rely on output types list
+    auto output_shapes = fifo_queue->get_component_shapes();
+    for (size_t ind = 0; ind < output_shapes.size(); ++ind) {
+        // update output shapes from FIFOQueue with batch dimension
+        auto& output_shape = output_shapes[ind];
+        output_shape.insert(output_shape.begin(), batch_dim);
+    }
+
+    // try to extract output shapes using _output_shapes attribute if exists
+    if (output_shapes.size() == 0) {
+        output_shapes = node.get_attribute<vector<PartialShape>>("_output_shapes", vector<PartialShape>{});
+    }
 
     OutputVector queue_dequeue_outputs;
     // perform auto-pruning: create Parameter nodes for each output of QueueDequeue
     for (size_t output_ind = 0; output_ind < output_num; ++output_ind) {
-        auto output_shape = output_shapes[output_ind];
+        auto output_shape = PartialShape::dynamic();
+        if (output_ind < output_shapes.size()) {
```

Comment:
auto output_shape = output_ind < output_shapes.size() ? output_shapes[output_ind] : PartialShape::dynamic();

Reply:
I will fix it in separate PR